spec/acceptance/store_proxy_spec.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require_relative '../spec_helper'
+
+describe Rack::Attack::StoreProxy do
+  describe ".build" do
+    if defined?(::Dalli)
+      it "uses DalliProxy to handle ActiveSupport MemCacheStore" do
+        store = ActiveSupport::Cache::MemCacheStore.new
+        assert_instance_of Rack::Attack::StoreProxy::DalliProxy, Rack::Attack::StoreProxy.build(store)
+      end
+    end
+    if defined?(::Dalli) && defined?(::ConnectionPool)
+      it "uses DalliProxy to handle ActiveSupport MemCacheStore with a connection pool" do
+        store = ActiveSupport::Cache::MemCacheStore.new(pool_size: 5)
+        assert_instance_of Rack::Attack::StoreProxy::DalliProxy, Rack::Attack::StoreProxy.build(store)
+      end
+    end
+    if defined?(::Redis)
+      it "uses RedisProxy to handle plain redis backends" do
+        store = Redis.new
+        assert_instance_of Rack::Attack::StoreProxy::RedisProxy, Rack::Attack::StoreProxy.build(store)
+      end
+    end
+    if defined?(::Redis) && defined?(::ActiveSupport::Cache::RedisCacheStore)
+      it "uses RedisCacheStoreProxy to handle plain ActiveSupport RedisCacheStore" do
+        store = ActiveSupport::Cache::RedisCacheStore.new
+        assert_instance_of Rack::Attack::StoreProxy::RedisCacheStoreProxy, Rack::Attack::StoreProxy.build(store)
+      end
+    end
+    if defined?(::Redis) && defined?(::ActiveSupport::Cache::RedisCacheStore) && defined?(::ConnectionPool)
+      it "uses RedisCacheStoreProxy to handle plain ActiveSupport RedisCacheStore with a connection pool" do
+        store = ActiveSupport::Cache::RedisCacheStore.new(pool_size: 2)
+        assert_instance_of Rack::Attack::StoreProxy::RedisCacheStoreProxy, Rack::Attack::StoreProxy.build(store)
+      end
+    end
+  end
```

Comment:
Not sure I would place a test like this in the `spec/acceptance` folder. The idea behind test cases in that folder is to test from a gem user perspective, as much as possible, as opposed to internal implementation details of the gem.

It would be awesome if we could convert this into something that actually tests what you're actually reporting as the problem in the description, which is that "throttle isn't working when the store is a pooled `MemCacheStore`".

It seems what you need is something very very similar in essence to [spec/acceptance/stores/active_support_redis_cache_store_pooled_spec.rb](https://github.com/kickstarter/rack-attack/blob/master/spec/acceptance/stores/active_support_redis_cache_store_pooled_spec.rb) but for `MemCacheStore` instead of `RedisCacheStore`.